src/hotspot/share/gc/serial/defNewGeneration.cpp

```diff
@@ -876,11 +871,15 @@ HeapWord* DefNewGeneration::allocate(size_t word_size, bool is_tlab) {
   return result;
 }
 
-HeapWord* DefNewGeneration::par_allocate(size_t word_size,
-                                         bool is_tlab) {
+HeapWord* DefNewGeneration::par_allocate(size_t word_size) {
   return eden()->par_allocate(word_size);
 }
 
+HeapWord* DefNewGeneration::expand_and_allocate(size_t size) {
+  // We don't attempt to expand the young generation (but perhaps we should.)
+  return allocate(size);
+}
```

Comment:
Previously, the virtual method `expand_and_allocate` is inherited from `Generation` so that we have such strange implementation. But now, you remove the related methods in `Generation`, I think you can remove the method `DefNewGeneration::expand_and_allocate` and use `DefNewGeneration::allocate` instead.